tasks/Module 4/task_2.py

```diff
@@ -0,0 +1,39 @@
+"""Task 2 of Module 4."""
+from pathlib import Path
+import os
+from tkinter.filedialog import askdirectory
+
+
+def filling_in_the_gaps():
+    """
+    Find all files with a given prefix,
+    in a single folder and fill in any gaps in the numbering.
+    """
+    dir_path = askdirectory(title='Select Folder with spam* files')
+    digit = 1
+    dozen = 0
+    hundred = 0
+
+    os.chdir(dir_path)
+
+    for file in os.listdir(dir_path):
```

Comment:
I just leave here advice from our program chat ;)

Please keep in mind that `os.listdir` or `os.scandir` return directory entries in arbitrary order, so make sure that your algorithm doesn't rely on that order. Otherwise you may get file names without gaps in numbering, but their content will be mixed  The simplest way to be safe is to sort input files by name: https://docs.python.org/3/howto/sorting.html

Reply:
it was hard to miss that, but i somehow did xD
changed to `for file in sorted(os.listdir(dir_path))`

Reply:
@GeorgiiDermenzhy did you commit this change? ;)

Reply:
@musk-ox, just finished all the local changes and pushed :)